### PR TITLE
refactor propulsion test with gmock

### LINKED ISSUE #270: Refactor propulsion with gmock

### PR DESCRIPTION
**Pull request by elheck**
closes #270 
Refactored all existing mocks for propulsion to use gmock mocks.

### CODE REVIEW

File: tests/mock_libraries/propulsion/letodar_2204_mock.hpp [hunk 1]
```diff
@@ -2,20 +2,19 @@
...
-  }
-  auto SetSpeedInPercent(const float speed) noexcept -> types::DriverStatus override {
-    return types::DriverStatus::OK;
-  }
+  explicit MotorMock(std::unique_ptr<propulsion::Esc>) : Motor(nullptr) {}
+  MOCK_METHOD(const float, GetCurrentSpeedInPercent, (), (const, noexcept, override));
+  MOCK_METHOD(types::DriverStatus, SetSpeedInPercent, (const float), (noexcept, override));
 };
+
+using LeTodar2204 = MotorMock;
```
Comment (ThorsHamster) [inline tests/mock_libraries/propulsion/letodar_2204_mock.hpp:18]: What is the purpose of this using?

File: tests/propulsion/letodar_2204_test.cpp [hunk 2]
```diff
@@ -1,46 +1,56 @@
...
+
 namespace {
 class Letodar2204Tests : public ::testing::Test {
  protected:
+  Letodar2204Tests() {}
   virtual void SetUp() {
-    unit_under_test_ = std::make_unique<propulsion::LeTodar2204>(std::make_unique<propulsion::Esc>(esc_));
+    ON_CALL(*esc_, GetMaxPulseDurationInMicroSeconds()).WillByDefault(Return(100));
+    ON_CALL(*esc_, GetMinPulseDurationInMicroSeconds()).WillByDefault(Return(10));
+    ON_CALL(*esc_, SetPulseDuration).WillByDefault(Return(types::DriverStatus::OK));
```
Comment (ThorsHamster) [inline tests/propulsion/letodar_2204_test.cpp:16]: For consistency you could remove '()' from GetMaxPulseDurationInMicroSeconds and GetMinPulseDurationInMicroSeconds. Or?

**Review by ThorsHamster — COMMENTED** [review]
Just some minor questions. Otherwise: Good job.

File: tests/mock_libraries/propulsion/letodar_2204_mock.hpp (hunk 1 above)
Comment (elheck) [reply tests/mock_libraries/propulsion/letodar_2204_mock.hpp:18]: This is a fancy c++11 way of making a typedef.
I need this here as my motorbuilder is expecting a class by this name, but I still want to inject my Mock object.

File: tests/propulsion/letodar_2204_test.cpp (hunk 2 above)
Comment (elheck) [reply tests/propulsion/letodar_2204_test.cpp:16]: You are correct.
Fixed with [this](https://github.com/Squiddrone/Squiddrone/pull/271/commits/0a5c06b2e31fe754073f4bf7810e6ee8f942a0d9) commit

**Review by ThorsHamster — APPROVED** [review]
LGTM

**Pull request merged**